src/Stratis.FederatedPeg.Features.FederationGateway/MonitorChain/OpReturnDataReader.cs

```diff
@@ -123,5 +124,23 @@ private static byte[] RemoveOpReturnOperator(byte[] rawBytes)
         {
             return rawBytes.Skip(2).ToArray();
         }
+
+        ///<inheritdoc />
+        public string TryGetTargetAddressFromOpReturn(Transaction transaction)
+        {
+            var opReturnAddresses = transaction.Outputs

```

Comment:
I am not sure we would have a case where there's more than one address; an OP_RETURN output simply can't fit very much data.

Reply:
But can you not get many op_returns out of one transaction? I think that is what happens in the matching test.
I think this is a potential for confusion in handling the transaction and that should help...